custom_components/bosch_alarm/config_flow.py

```diff
@@ -50,14 +52,30 @@ async def try_connect(hass: HomeAssistant, data: dict[str, Any]):
     Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
     """
     panel = Panel(host=data[CONF_HOST], port=data[CONF_PORT],
-                  passcode=data[CONF_PASSWORD])
+                  automation_code=data[CONF_PASSWORD], installer_code=data.get(CONF_INSTALLER_CODE, None))
     try:
         await panel.connect(Panel.LOAD_BASIC_INFO)
     finally:
         await panel.disconnect()
 
+    errors = {}
+
+    try:
+        await panel.connect(Panel.LOAD_BASIC_INFO)
+    except (PermissionError, ValueError):
+        errors["base"] = "invalid_auth"
+    except (OSError, ConnectionRefusedError, ssl.SSLError, asyncio.exceptions.TimeoutError):
+        errors["base"] = "cannot_connect"
+    except Exception:  # pylint: disable=broad-except
+        _LOGGER.exception("Unexpected exception")
+        errors["base"] = "unknown"
+    finally:
+        await panel.disconnect()
+    
+    if errors:
+        return (errors, None, None)
     # Return info that you want to store in the config entry.
-    return (panel.model, panel.serial_number)
+    return (None, panel.model, panel.serial_number)
```

Comment:
This is rather awkward.
Can we have this function continue to return (model, serial), and throw the string to be populated into errors as an exception?